smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpProtocolGeneratorUtils.java

```diff
@@ -357,7 +357,7 @@ static Set<StructureShape> generateUnifiedErrorDispatcher(
 
                 // Get the protocol specific error location for retrieving contents.
                 String errorLocation = bodyErrorLocationModifier.apply(context, "parsedBody");
-                writer.openBlock("return throwDefaultError({", "})", () -> {
+                writer.openBlock("return throwDefaultError({", "}) as never", () -> {
```

Comment:
this is easier than releasing a change to the return type of `throwDefaultError`, which may cause instability in existing clients